Move the cancel api calls to the end of the survey

dzver opened the pull request:
Part of DOTCOM-13834

## Proposed Changes

* It was decided that the cancel api calls should go back to the end of the survey again
* This will happen without reverting the last 5-6 PRs and preserving most of the changes done in them, like removing unrelated steps
* It restores the cancellation solutions to the state from right before, including the downgrade Premium to Personal
* It introduces a Skip button that skips the survey and goes straight to cancellation

## Why are these changes being made?

* Part of the broader Cancel Survey redesign + regulation requirements

## Testing Instructions

You have to test each one of the following
* Get a plan without a domain
* Get a plan with a domain
* Get a domain without a plan
* Buy an add-on like space
* Get Jetpack Boost or another Jetpack/Akismet upgrade

Then go ahead and cancel them.

For the plan+domain, go over the cancellation solutions one by one and see if they match your expectations

Highlights:
* With an annual Premium plan, selecting too expensive gives you the downgrade to Personal option
* With an annual Personal plan, selecting too expensive gives you a downgrade to monthly

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

dzver committed:
WIP: Move the cancel/refund API calls back to the end

dzver committed:
Fix the Jetpack survey

dzver committed:
Busy button in Jetpack Cancel Survey

Also, the intermediate step - it got back for refundable plans, and
shouldn't have

dzver committed:
Remove comment

dzver committed:
Add skip button

dzver committed:
One more skip instead of skip survey

dzver committed:
Revert get-upsell-type and upsell-step to commits from before

the api reordering

dzver committed:
Restore the downgrade

fditrapani commented:
Nice work @dzver! Functionally this works well for most cases I tested.

One of the exceptions was when I cancel a domain without a plan or anything. In production, we show a custom survey for domains. In the PR, I go straight to the second survey without the initial questions.

Along with these functional changes, can we also make the following updates:
1. We discussed removing the skip button for now. We'll want to test that option at a later stage so we can measure its impact better.

2. For the first survey, can we change the Submit button label to Continue? 
<img width="1822" height="1191" alt="image" src="https://github.com/user-attachments/assets/565632fd-f474-47f8-aaa3-8a443e86344d" />

dzver committed:
Unbreak the domain cancel flow

dzver committed:
Use the component button for consistent busy behavior

dzver committed:
Remove skip button per request

dzver committed:
Use the Gutenberg Button instead of Automattic

dzver committed:
Change the first step button to 'Continue'

dzver commented:
Changes:
- Fixed a regression not showing the Domain Cancellation flow
- I removed the Skip button
- Changed the button of the First step in the default flow to be 'Continue'
- Used the WordPress button instead of just button so the busy state in Domain Cancellation works (didn't matter before because there was no busy state)

dzver committed:
Don't skip the survey if marketplace subscriptions exist

dzver committed:
Prevent double display of the Marketplace dialog

fditrapani reviewed (approved):
Thanks for the updates! Looks good to me now. 

> I think we can incorporate the information on this dialog into the Atomic screen as a separate change

Makes sense. I created a linear issue for it and assigned myself to it so I can look into it when I'm back from AFK.
